spec/features/prolong_token_build_spec.rb

```diff
@@ -0,0 +1,173 @@
+
+      choose(contact_metier_option) if api == 'entreprise'
+      choose(contact_technique_option)
+      click_button('next_prolong_token_wizard')
+    end
+
+    before do
+      login_as(authenticated_user)
+    end
+
+    describe 'when contact_technique or contact_metier is not present' do
+      let(:in_charge_option) { 'prolong_token_wizard_owner_still_in_charge' }
+      let(:project_purpose_option) { 'prolong_token_wizard_project_purpose_true' }
+
+      it 'does not allow to keep contacts' do
+        go_to_prolong_token_start
+
+        choose(in_charge_option)
+        click_button('next_prolong_token_wizard')
+
+        choose(project_purpose_option)
+        click_button('next_prolong_token_wizard')
+
+        expect(page).to have_no_css('#prolong_token_wizard_contact_metier_true')
+        expect(page).to have_no_css('#prolong_token_wizard_contact_technique_true')
+
+        expect(page).to have_css('#prolong_token_wizard_contact_metier_false')
+        expect(page).to have_css('#prolong_token_wizard_contact_technique_false')
+      end
+    end
+
+    describe 'when on api entreprise' do
+      let!(:authenticated_user) { create(:user, :demandeur, :contact_metier, :contact_technique) }
+
+      let!(:authorization_request) do
+        create(
+          :authorization_request,
+          :with_demandeur,
+          :with_contact_metier,
+          :with_contact_technique,
+          demandeur: authenticated_user,
+          contact_metier: authenticated_user,
+          contact_technique: authenticated_user,
+          api:,
+          status: 'validated'
+        )
+      end
+
+      describe 'fills the wizard' do
+        let(:in_charge_option) { 'prolong_token_wizard_owner_still_in_charge' }
+
+        context 'when we valid all questions' do
+          let(:project_purpose_option) { 'prolong_token_wizard_project_purpose_true' }
+          let(:contact_metier_option) { 'prolong_token_wizard_contact_metier_true' }
+          let(:contact_technique_option) { 'prolong_token_wizard_contact_technique_true' }
+
+          it 'prolongs the token' do
+            fill_wizard
+
+            click_link('finished_prolong_token_wizard')
+
+            expect(page).to have_content('Votre jeton a été prolongé')
+            expect(token.reload.exp).to eq(18.months.from_now.to_i)
+          end
+        end
+
+        context 'when at least one field has to be changed' do
+          let(:project_purpose_option) { 'prolong_token_wizard_project_purpose_false' }
+          let(:contact_metier_option) { 'prolong_token_wizard_contact_metier_false' }
+          let(:contact_technique_option) { 'prolong_token_wizard_contact_technique_false' }
+
+          it 'follows update path' do
+            fill_wizard
+
+            expect(page).to have_content("de la raison d'être du projet")
+
+            expect(token.reload.exp).to eq(exp)
+          end
+        end
+      end
+    end
+
+    describe 'when on api particulier' do
```

Comment:
```suggestion
    describe 'when on api particulier', app: :api_particulier do
```

d'ailleurs je trouve ça étrange que cela fonctionne, on peut démarrer un wizard API P dans API E ?

Reply:
imo c'est pas très grave vu que les ids sont des uuids, simple remarque ^^